File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/10 - 2 and 3 May 2019/04. Tourist Shop.py
budget = float(input())

command = input()

counter = 0
total = 0

while command != "Stop":
    product_price = float(input())
    counter += 1
    if counter % 3 == 0:
        product_price *= 0.5
    if product_price > budget:
        print("You don't have enough money!")
        print(f"You need {(product_price - budget):.2f} leva!")
        break
    budget -= product_price
    total += product_price
    command = input()

if command == "Stop":
    print(f"You bought {counter} products for {total:.2f} leva.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/02.  Online Retake Exam - 2 and 3 May 2019/04. Tourist Shop.js
// Късо линкче: https://git.io/JfvMz

function touristShop(input) {
    // Mask
    let budget = Number(input.shift());
    let products = 0;
    let total = 0;

    while('I am an idiot!') {
        let product = input.shift(); 
        if (product == 'Stop') {
            console.log(`You bought ${products} products for ${total.toFixed(2)} leva.`);
            break;
        }
        products += 1;
        let price = Number(input.shift());
        if (products % 3 == 0) {
            price = price / 2;
        }

        if (budget - price < 0) {
            console.log(`You don't have enough money!`);
            console.log(`You need ${(price - budget).toFixed(2)} leva!`);
            break;
        }
        
        total += price;
        budget -= price;
    }
}

touristShop([153.20, 'Backpack', 25.20, 'Shoes', 54, 'Sunglasses', 30, 'Stop']);  // Should return:
// You bought 3 products for 94.20 leva.

touristShop([54, 'Thermal underwear', 24, 'Sunscreen', 45]);  // Should return:
// You don't have enough money!
// You need 15.00 leva!
